gadgets/src/integers/uint/arithmetic/div.rs

```diff
@@ -54,100 +56,89 @@ macro_rules! div_int_impl {
                 //   end
                 // end
 
-                if other.eq(&Self::constant(0 as <$gadget as Integer>::IntegerType)) {
-                    return Err(SynthesisError::DivisionByZero.into());
+                // Enforce D != 0
```

Comment:
Explain: This gadget is also completely rewritten, which avoids the incorrect constant allocations.